pyomo/contrib/doe/examples/reactor_compute_FIM.py

```diff
@@ -0,0 +1,105 @@
+#  Initiative (CCSI), and is copyright (c) 2022 by the software owners:
+#  TRIAD National Security, LLC., Lawrence Livermore National Security, LLC.,
+#  Lawrence Berkeley National Laboratory, Pacific Northwest National Laboratory,
+#  Battelle Memorial Institute, University of Notre Dame,
+#  The University of Pittsburgh, The University of Texas at Austin,
+#  University of Toledo, West Virginia University, et al. All rights reserved.
+#
+#  NOTICE. This Software was developed under funding from the
+#  U.S. Department of Energy and the U.S. Government consequently retains
+#  certain rights. As such, the U.S. Government has been granted for itself
+#  and others acting on its behalf a paid-up, nonexclusive, irrevocable,
+#  worldwide license in the Software to reproduce, distribute copies to the
+#  public, prepare derivative works, and perform publicly and display
+#  publicly, and to permit other to do so.
+#  ___________________________________________________________________________
+
+
+import numpy as np
+import pyomo.common.unittest as unittest
+from pyomo.contrib.doe.examples.reactor_kinetics import create_model, disc_for_measure
+from pyomo.contrib.doe import (
+    DesignOfExperiments,
+    MeasurementVariables,
+    calculation_mode,
+    DesignVariables,
+    finite_difference_step,
+)
+
+
+def main():
+    ### Define inputs
+    # Control time set [h]
+    t_control = [0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1]
+    # Define parameter nominal value
+    parameter_dict = {'A1': 85, 'A2': 370, 'E1': 8, 'E2': 15}
+
+    # Define measurement object
+    measurements = MeasurementVariables()
+    measurements.add_variables("C", # measurement object
+                               indices={0: ['CA', 'CB', 'CC'], 1: t_control}, # 0,1 are indices of the index sets 
+                               time_index_position=1)
+
+    # design object
+    exp_design = DesignVariables()
+
+    # add CAO as design variable
+    exp_design.add_variables(
+        "CA0",
+        indices={0: [0]},
+        time_index_position=0,
+        values= [5],
+        lower_bounds=1,
+        upper_bounds=5,
+    )
+
+    # add T as design variable
+    exp_design.add_variables(
+        'T',
+        indices= {0: t_control},
+        time_index_position=0,
+        values=[570, 300, 300, 300, 300, 300, 300, 300, 300], # same length with t_control
+        lower_bounds=300,
+        upper_bounds=700,
+    )
+
+    ### Test sequential_finite mode
+    sensi_opt = calculation_mode.sequential_finite
+
+    doe_object = DesignOfExperiments(
+        parameter_dict,
+        exp_design,
+        measurements,
+        create_model,
+        discretize_model=disc_for_measure,
+    )
+
+    result = doe_object.compute_FIM(
+        mode=sensi_opt,
+        scale_nominal_param_value=True,
+        formula=finite_difference_step.central,
+    )
+
+    result.calculate_FIM(doe_object.design_values)
```

Comment:
What is the difference between compute_FIM and caculate_FIM? Can we rename one to make the distinction more clear?

Reply:
How about change result.calculate_FIM to result.calculate_FIM_from_jacobian? The function is calculating FIM from a given jacobian information.